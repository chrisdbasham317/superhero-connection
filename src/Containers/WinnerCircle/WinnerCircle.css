.section--winner-circle {
  align-items: center;
  background: url('https://ae01.alicdn.com/kf/HTB1fC1eXyzxK1RjSspjq6AS.pXaU/Laeacco-Stage-Backgrounds-Shiny-Spotlight-Podium-Show-Red-Carpet-Polka-Dots-Kid-Portrait-Photo-Backdrops-Photocall.jpg_q50.jpg');
  background-position-y: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  flex-direction: column;
  height: 92vh;
}

.h1--winner-circle{
  font-size: 1.8rem;
  margin: 0;
}

.h2--winner-name {
  text-align: center;
}

.img--winner-img {
  border: 8px outset darkslategrey;
  height: 55vh;
}