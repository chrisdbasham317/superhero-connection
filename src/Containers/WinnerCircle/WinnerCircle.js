import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './WinnerCircle.css'

export const WinnerCircle = ({winner, heroes}) => {
  const winnerInfo = heroes.filter(hero => hero.name === winner)[0]
  return (
    <section className='section--winner-circle'>
    {!winner && <h1>It seems those heroes are evenly matched</h1>}
      {winner &&
      <section>
        <h1 className='h1--winner-circle'>WinnerCircle</h1>
        <div className='div--winner-info'>
          <h2 className='h2--winner-name'>{winnerInfo.name}</h2>
          <img className='img--winner-img'src={ winnerInfo.images.md } alt='Winner' />
        </div>
      </section>}
    </section>
  )
}

export const mapStateToProps = ({winner, heroes}) => ({
  winner,
  heroes
})

export default connect(mapStateToProps)(WinnerCircle);

WinnerCircle.propTypes = {
  winner: PropTypes.string,
  heroes: PropTypes.array
}