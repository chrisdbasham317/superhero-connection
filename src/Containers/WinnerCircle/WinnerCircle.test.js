import React from 'react';
import { shallow } from 'enzyme';
import { WinnerCircle, mapStateToProps } from './WinnerCircle';

describe('WinnerCircle', () => {
  let wrapper
  const mockHeroes = [
    {
      id: 4,
      name: 'Superman',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 100,
      }
    },
    {
      id: 5,
      name: 'superboy',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 85,
      }
    }
  ]
  beforeEach(() => {
    wrapper = shallow(<WinnerCircle
      winner={'Superman'}
      heroes={mockHeroes}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return an object with the heroes array, and the winner', () => {
    const expected = {
      heroes: mockHeroes,
      winner: 'Superman'
    };
    const mockState = {
      heroes: mockHeroes,
      winner: 'Superman'
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});