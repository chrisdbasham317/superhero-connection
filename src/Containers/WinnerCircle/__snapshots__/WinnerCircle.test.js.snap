// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerCircle should match the snapshot 1`] = `
<section
  className="section--winner-circle"
>
  <section>
    <h1
      className="h1--winner-circle"
    >
      WinnerCircle
    </h1>
    <div
      className="div--winner-info"
    >
      <h2
        className="h2--winner-name"
      >
        Superman
      </h2>
      <img
        alt="Winner"
        className="img--winner-img"
        src="medium image"
      />
    </div>
  </section>
</section>
`;
